src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/MethodToStateMachineRewriter.cs

```diff
@@ -365,7 +369,7 @@ private BoundStatement PossibleIteratorScope(ImmutableArray<LocalSymbol> locals,
                 }
                 else
                 {
-                    foreach (var field in ((CapturedToExpressionSymbolReplacement)proxy).HoistedFields)
+                    foreach (var field in ((CapturedToExpressionSymbolReplacement<StateMachineFieldSymbol>)proxy).HoistedSymbols)
```

Comment:
>field [](http://example.com/codeflow?start=33&length=5)

Naming and/or consider adding an assertion that the symbols are fields

Reply:
The cast will fail if they're not all fields.